kodi_addon_checker/addons/Repository.py

```diff
@@ -0,0 +1,48 @@
+"""
+    Copyright (C) 2018 Team Kodi
+    This file is part of Kodi - kodi.tv
+
+    SPDX-License-Identifier: GPL-3.0-only
+    See LICENSES/README.md for more information.
+"""
+
+from .Addon import Addon
+
+import gzip
+import requests
+import xml.etree.ElementTree as ET
+from io import BytesIO
+
+
+class Repository(object):
+    """docstring for Repository"""
+    def __init__(self, version, path):
+        super(Repository, self).__init__()
+        self.version = version
+        self.path = path
+        gz_file = requests.get(path, timeout=(10, 10)).content
```

Comment:
maybe you can check the header first instead of assuming it is gzip

Reply:
This requests an explicit gz file and doesn't assume auto compression by the web server.
https://github.com/xbmc/addon-check/blob/b803624f41a8d0e84629063f3fc886d86ae0f458/kodi_addon_checker/check_addon.py#L29